vsphere/resource_vsphere_compute_cluster.go

```diff
@@ -531,7 +540,7 @@ func resourceVSphereComputeCluster() *schema.Resource {
 			"vsan_unmap_enabled": {
 				Type:        schema.TypeBool,
 				Optional:    true,
-				Default:     false,
+				Computed:    true,
```

Comment:
Is this computed? does it get automatically toggled under certain conditions by vsphere? If so, what restrictions would be in place for the user setting a discrete value (setting it false but vsphere wants it to be true?)

Reply:
Yes, this feature is related with vSAN ESA switch, as mentioned in doc file, vSAN unmap will be by default enabled along with vSAN ESA. Disabling unmap separately is not allowed in vSAN ESA cluster.

Reply:
I'm going to have to wrap my head around this subtle change. Historically there exists a problem around Optional + Computed and Booleans, and I'm unsure if this would present a subtle breaking change.

Current behavior:

* User specifies nothing OR false
    * Defaults false (or explicitly false) --> unmap is disabled
* User specifies true
    * unmap is enabled
* User DELETES the attribute
   * Because of the default, semantically this is the same as setting back to false --> unmap is disabled

New behavior:

* User specifies nothing
    * If vSAN ESA is disabled --> unmap is also disabled
    * If vSAN ESA is enabled --> unmap is also enabled
* User specified true
    * If vSAN ESA is disabled --> no problem, unmap is now enabled
    * If vSAN ESA is enabled ---> no problem, redundant, unmap remains enabled
* User specified false
    * If vSAN ESA is disabled --> no problem, unmap is disabled
    * If vSAN ESA is enabled --> PROBLEM, I assume the API request to set this to false will error? Safest bet would be to prevent this in the Create/Update codepath, unfortunately with the legacy provider SDK, cross-attribute value validation has to be done at apply time (can't be done in Validate funcs/plantime).
* User DELETES the attribute
    * Terraform takes this as set to `nil`, however since Computed is true, Terraform just forwards along whatever was previously set in State, therefore no change is triggered (no longer gets set back to false).
    
This is a very nuanced situation, I'd argue it's probably not such a hard deviation of behavior that it would a dealbreaker. A user upset could explicitly try and set it to false, and probably experience an error (either from the vSphere API itself, or if we codify this restriction into the implementation).

We could allow this schema change and deal with any "fallout" if there's an edgecase I failed to capture (Boolean + Optional + Computed has historically created extremely frustrating scenarios so there could be one I'm forgetting).

Or we could not switch to Computed and simply error during apply if the user has it unset/set to false IF vSAN ESA is enabled.

What are your thoughts? I actually lean towards my latter suggestion, I have always hated the semantic ambiguity of unset + computed potentially resulting in `True` on the backend, I much prefer `nil` be a semantic `False` assumption throughout codebases.

Reply:
Regarding this `vsan_unmap_enabled ` property, since it has to be enabled on ESA cluster and cannot be disabled until ESA is enabled, should we introduce it to the user at all ? Is there a case when the configuration might be `vsan_esa_enabled =false` and `vsan_unmap_enabled=true` ? If so would not it be better to introduce a flag attribute how the `vsan_unmap_enabled` should be handled if `vsan_esa_enabled` is turned off, somethng like  `turn off` | `keep as is`? 

Considering this as a solution to above comments because the feature is vSAN ESA and this is what we want to introduce to the users and I am not sure about the part of the `vsan_unmap_enabled`. If it is important feature like we want to give the users control over the `vsan_unmap_enabled` attribute and It may create a combination with the `vsan_esa_enabled`  which will not work for the user, will make terraform end up with improper state or sth probably a validation should be added in the beginning of create and update methods of the resource to notify the user that the created configuration is not correct.

Reply:
The constraints between vsan esa and vsan unmap are:
1. vsan unmap will be enabled by default when users try to enable vsan esa.
2. vsan unmap could not be disabled separately with vsan esa enabled.
3. vsan unmap could be set as true standalone only with vsan enabled, that's the case `vsan_esa_enabled =false` and `vsan_unmap_enabled=true`
In latest commit, we modify vsan unmap from computed true to false back, and make it default value as false. Then only one invalid combination `vsan_esa_enabled =true` and `vsan_unmap_enabled=false`is forbidden and terraform will directly output err msg. It also means users should explicitly configure both vsan unmap and vsan esa as true if they're going to enable vsan esa, this constraint is added in Line 1484. We could have an offline discussion if there're any other concerns, thanks.